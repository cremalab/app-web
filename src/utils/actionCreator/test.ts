import { actionCreator } from "."

export interface User {
  name: string
}

enum ActionType {
  UserAdd = "UserAdd",
  Sum = "Sum",
}

describe("actionCreator", () => {
  it("returns action creator where argument is passed as action payload", () => {
    // Arrange
    const actionItemAdd = actionCreator<ActionType.UserAdd, [User], User>(
      ActionType.UserAdd,
    )

    // Act
    const action = actionItemAdd({ name: "Jeff" })

    // Assert
    expect(action).toMatchSnapshot()
  })

  it("returns action creator where arguments are converted to payload", () => {
    // Arrange
    const sum = (a: number, b: number) => a + b
    const actionSum = actionCreator(ActionType.Sum, sum)

    // Act
    const action = actionSum(2, 3)

    // Assert
    expect(action).toMatchSnapshot()
  })
})
