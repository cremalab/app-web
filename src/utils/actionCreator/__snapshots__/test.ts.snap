// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionCreator returns action creator where argument is passed as action payload 1`] = `
Object {
  "payload": Object {
    "name": "Jeff",
  },
  "type": "UserAdd",
}
`;

exports[`actionCreator returns action creator where arguments are converted to payload 1`] = `
Object {
  "payload": 5,
  "type": "Sum",
}
`;
